supervisor/supervisor.go

```diff
@@ -48,9 +48,6 @@ func New(stateDir string, runtimeName, shimName string, runtimeArgs []string, ti
 	}
 	go s.exitHandler()
 	go s.oomHandler()
-	if err := s.restore(); err != nil {
```

Comment:
@keloyang before hand, request to the non health GRPC API would fail as the service is not registered. Now, it'll deadlock unless the context has a timeout or is cancelled if `restore()` itself has an issue.

It's a tiny difference, but it's there nonetheless.

Anyway, let's merge this for now.